src/app/services/comment.service.ts

```diff
@@ -43,13 +47,15 @@ export class CommentService {
     await firebase.database().ref(`/blog/users/${userId}`).set(user);
   }
 //finds all comments based in the parentId of the comment
-  getCommentsByParentId(parentId, type) {
+  commentsByParentIdRef(parentId, type) {
     const result = firebase.database().ref(`/blog/${type}/${parentId}`);
-    return result;
+    if(result){
```

Comment:
This isn't well-known but my understanding is that a firebase database reference will always return a result, so we don't actually need to check if the reference is there.